src/memory.rs

```diff
@@ -0,0 +1,153 @@
+//! Utilities for converting Kubernetes quantities to Java heap settings.
+
+use k8s_openapi::apimachinery::pkg::api::resource::Quantity;
+
+use crate::error::{Error, OperatorResult};
+use std::str::FromStr;
+
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
+pub enum BinaryMultiple {
+    Kibi,
+    Mebi,
+    Gibi,
+    Tebi,
+    Pebi,
+    Exbi,
+}
+
+impl BinaryMultiple {
+    pub fn to_legacy(&self) -> String {
+        match self {
+            BinaryMultiple::Kibi => "k".to_string(),
+            BinaryMultiple::Mebi => "m".to_string(),
+            BinaryMultiple::Gibi => "g".to_string(),
+            BinaryMultiple::Tebi => "t".to_string(),
+            BinaryMultiple::Pebi => "p".to_string(),
+            BinaryMultiple::Exbi => "e".to_string(),
+        }
+    }
+
+    pub fn upscale(&self) -> Self {
+        match self {
+            BinaryMultiple::Kibi => BinaryMultiple::Kibi,
+            BinaryMultiple::Mebi => BinaryMultiple::Kibi,
+            BinaryMultiple::Gibi => BinaryMultiple::Mebi,
+            BinaryMultiple::Tebi => BinaryMultiple::Gibi,
+            BinaryMultiple::Pebi => BinaryMultiple::Tebi,
+            BinaryMultiple::Exbi => BinaryMultiple::Pebi,
+        }
+    }
+}
+
+impl FromStr for BinaryMultiple {
+    type Err = Error;
+
+    fn from_str(q: &str) -> OperatorResult<BinaryMultiple> {
+        let lq = q.to_lowercase();
+        match lq.as_str() {
+            "ki" | "kib" => Ok(BinaryMultiple::Kibi),
```

Comment:
I've done a bit of reading and experimenting and as far as I can tell K8s also accepts m | "" | k | M | G | T | P | E as suffix here, which are a different unit, but seem to be treated as equal by k8s at least for PV(C) sizing.
I think we can probably get away with just adding these as synonyms in this list and treating them as equal?